src/include/simeng/CoreInstance.hh

```diff
@@ -139,6 +137,10 @@ class CoreInstance {
 
   /** Reference to the global memory pointer */
   std::shared_ptr<simeng::memory::Mem> memory_ = nullptr;
+
+  /** Function, passed to the Core class, to communicate a syscall generated by
+   * an exception handler to the system's syscall handler. */
+  std::function<void(const simeng::OS::SyscallInfo)> syscallHandle_;
```

Comment:
Would it be better to typedef the `std::function<void(const simeng::OS::SyscallInfo)>` as SyscallHandle and then rename the memberaccordingly? 
I think giving verbose types to the callback function helps communicate their intent clearly.

Reply:
Made the change, good idea